### DISCUSSION TITLE
Anyway to override default rpc hanlder methods ?

### DISCUSSION THREAD

**Discussion by itsyaasir** [paradigmxyz/reth#8294] [Help]
Hi, I am currently working on a forked reth and I needed a way to override the default rpc method handler; for example, I want to override the default `eth_estimateGas` or `eth_call` with custom implementation, is there anyway to do this ? I have seen addition of custom RPC endpoints but I don't want to add new one just override it. 

Thanks

**Comment by mattsse** [comment]
we definitely want to support that, currently this is not possible, but should be shortly

**Comment by itsyaasir** [comment]
Thank you; Will keep track of the issue

**Comment by emhane** [comment]
possible now, checkout how it's done for optimism https://github.com/paradigmxyz/reth/tree/main/crates/optimism/rpc/src/eth

**Comment by itsyaasir** [comment]
I have verified this solution, which partially addresses the issue I am facing. I was seeking a resolution similar to the raw transaction forwarder. My intention was to direct certain calls to my node . Unfortunately, the current override implementation does not support the initialization of a client with an RPC URL. It is possible that your team has integrated this feature, but I have been unable to locate on how to do this. Some help will be appreciated, thanks.

To give more context; I am trying to do something like this for the `gas_price`:
```rust
    async fn gas_price(&self) -> Result<U256, Self::Error> {
         let client = EthJsonRpcClient::new(ReqwestClient::new(url.into())); // The url needs to be accessed and will be passed in the cli. 
        let gas_price = client
            .gas_price()
            .await
            .map_err(|err| EthApiError::EvmCustom(err.to_string()))?;
        Ok(U256::from(gas_price.as_u128()))
    }
    ```

**Comment by emhane** [comment]
ah ok, I follow. @mattsse mentioned some work by @prestwich recently that should enable this, right guys?

**Reply by prestwich** [reply]
My PR is stuck in limbo:
https://github.com/paritytech/jsonrpsee/pull/1387

Another PR was merged that allows removing things from modules 
https://github.com/paritytech/jsonrpsee/pull/1416
